src/components/record/related-table-actions.tsx

```diff
@@ -519,29 +510,15 @@ const RelatedTableActions = ({
       setShowPureBinarySpinner(true);
 
       validateSessionBeforeMutation(() => {
-        const deleteResponse = (response: any) => {
-          setShowPureBinarySpinner(false);
-
-          // Show modal popup summarizing total # of deletions succeeded and failed
-          response.clickOkToDismiss = true;
-
-          // TODO: - improve partial success and use TRS to check delete rights before giving a checkbox
-          //       - some errors could have been because of row level security
-          dispatchError({
-            error: response,
-            isDismissible: true,
-            closeBtnCallback: () => {
-              // ask recordset to update the modal
-              if (!!containerRef.current) {
-                // NOTE: This feels very against React but the complexity of our flow control provider seems to warrant doing this
-                fireCustomEvent(CUSTOM_EVENTS.FORCE_UPDATE_RECORDSET, containerRef.current, {
-                  cause: LogReloadCauses.ENTITY_BATCH_UNLINK,
-                  pageStates: { updateResult: true, updateCount: true, updateFacets: true },
-                  response: response,
-                });
-              }
-            },
-          });
+        const deleteResponse = () => {
+          leafReference
+            .deleteBatchAssociationTuples(relatedModel.recordsetProps.parentTuple, selectedRows)
+            .then(()=>{
+              setUnlinkPureBinaryModalProps(null);
+              updateRecordPage(true, LogReloadCauses.RELATED_BATCH_UNLINK);
+            })
+            .catch(deleteError)
```

Comment:
Since this was refactored, `deleteError` is only being called in this one place. Either undo the change to `deleteResponse` so that both `deleteResponse` and `deleteError` are used twice or put the contents of `deleteError` in the catch here.